### PR TITLE
Add enum type

### PR DESCRIPTION
**Pull request by mamiksik**

### CODE REVIEW

**Comment by filak-sap** [comment]
Good job. Fix the pylint warnings please:

```
pyodata/v2/model.py:1111: [R1705(no-else-return), EnumType.__getattr__] Unnecessary "else" after "return"
pyodata/v2/model.py:1119: [R1705(no-else-return), EnumType.__getitem__] Unnecessary "else" after "return"
pyodata/v2/model.py:1156: [W0622(redefined-builtin), EnumType.from_etree] Redefining built-in 'type'
```

By the way, users get use `getattr` to get members with white space. You can also enhance __getitem__ to support string parameters and call __getattr__ for str items.

**Comment by mamiksik** [comment]
"pyodata/v2/model.py:1127:5: E303 too many blank lines (2)". It's strange since pylint on my pc has not found any problems.

**Comment by jfilak** [comment]
> "pyodata/v2/model.py:1127:5: E303 too many blank lines (2)". It's strange since pylint on my pc has not found any problems.

Check out pylint version.

**Comment by mamiksik** [comment]
I fixed the warning. My pylint version is 2.3.1.

File: pyodata/v2/model.py [hunk 1]
```diff
@@ -620,6 +646,21 @@ def complex_type(self, type_name, namespace=None):
...
+            except KeyError:
+                raise KeyError(f'EnumType {type_name} does not exist in Schema Namespace {namespace}')
+
+        for decl in list(self._decls.values()):
+            try:
+                return decl.enum_types[type_name]
+            except KeyError:
+                pass
+
+        raise KeyError(f'ComplexType {type_name} does not exist in any Schema Namespace')
```
Comment (filak-sap) [inline pyodata/v2/model.py]: s/ComplexType/EnumType/

**Pull request merged**